src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
+
+  let sum = 0;
+
+  men.forEach((m) => sum += m.died - m.born);
+
+  return sum/men.length;
```

Comment:
The `forEach` loop you're using to calculate the sum of ages can be replaced with the `reduce` method. This will allow you to perform the sum calculation within the same loop that is filtering the data, rather than iterating over the array again.

```suggestion
  return men.reduce((sum, man) => sum + man.died - man.born, 0) / men.length;
```